src/cljs/webchange/dashboard/classes/class_modal/views.cljs

```diff
@@ -1,11 +1,55 @@
 (ns webchange.dashboard.classes.class-modal.views
   (:require
-    [cljs-react-material-ui.reagent :as ui]
-    [re-frame.core :as re-frame]
-    [reagent.core :as r]
-    [webchange.dashboard.classes.subs :as classes-subs]
-    [webchange.dashboard.students.subs :as students-subs]
-    [webchange.dashboard.classes.events :as classes-events]))
+   [cljs-react-material-ui.reagent :as ui]
+   [re-frame.core :as re-frame]
+   [reagent.core :as r]
+   [webchange.dashboard.classes.subs :as classes-subs]
+   [webchange.dashboard.students.subs :as students-subs]
+   [webchange.dashboard.classes.events :as classes-events]))
+
+(def courses
+  [{:course-id 4 :name "english"}
+   {:course-id 2 :name "spanish"}])
+
+(defn- get-styles
+  []
+  {:main        {:width          "120px"
+                 :margin         "0 20px"}
+   :div-course  {:margin-bottom  "5px"
+                 :margin-top     "30px"}
+   :spn-course  {:margin-left    "20px"
+                 :font-family    "Roboto, Helvetica, Arial, sans-serif"
+                 :color          "#595959"}})
+
+(defn- menu-item
+  [{:keys [course-id name]}]
+  [ui/menu-item {:key course-id :value name}
+   [ui/typography {:inline  true
+                   :variant "inherit"}
+    name]])
+
+(defn get-course-id
+  [props]  (->> courses
+                (filter #(= props (:name %)))
+                (first)
+                (:course-id)))
+
+(defn get-course-name
+  [props]  (->> courses
+                (filter #(= props (:course-id %)))
+                (first)
+                (:name)))
+
+(defn course-selector [props]
+  (let [styles (get-styles)]
+    [:div {:style  (:div-course styles)}
+     [:span {:style  (:spn-course styles)} "Course"]
+     [ui/select {:value     (get-course-name (:course-id @props))
+                 :variant   "outlined"
+                 :on-change #(swap! props assoc :course-id (get-course-id (->> % .-target .-value)))
```

Comment:
I suppose it would be better if `course-selector` get `value` and `on-change` handler. The rest logic of getting course name and mutating r/atom data keep in single component (`class-modal`)

Reply:
I didn't understand this point clearly, can you please explain me with an example

Reply:
I mean do not mutate external atom - it could be hard to understand in future. And separate logic and view if it's possible.

```
(defn class-modal
  []
  (let [class-data (r/atom (if (= :edit class-modal-state)
                             current-class
                             {:course-id 4 :name nil}))
        ...
        current-course (get-course-name (:course-id class-data))
        handle-course-changed #(swap! class-data assoc :course-id (get-course-id %))]
    ...
    [course-selector {:value     current-course
                      :on-change handle-course-changed}]
    ...))
```

```
(defn- course-selector
  [{:keys [value on-change]}]
  (let [styles (get-styles)]
    [:div {:style (:div-course styles)}
     [:span {:style (:spn-course styles)} "Course"]
     [ui/select {:value     value
                 :variant   "outlined"
                 :on-change #(on-change (->> % .-target .-value))
                 :style     (:main styles)}
      (for [course courses]
        (menu-item course))]]))
```

Reply:
I have added the handle-course-change method in the class-modal component. The handler is passed to the course-selector component as an argument as suggested. Please check & provide your reviews. Thanks!